### DISCUSSION TITLE
How to create a stiffening member in the IOM?

### DISCUSSION THREAD

**Discussion by BouwenmetStaal** [idea-statica/ideastatica-public#144] [General Discussion]
Hi all,

Is it possible to create a stiffening member in the IOM?
I could not find a method to add this to the OpenModel.

Could you provide me with a short example similar to the Cut beam by beam example found in the screenshot below?
![image](https://user-images.githubusercontent.com/48355355/164244665-c0a12e7d-3a3a-45f7-9ceb-2be58c086b0b.png)

![image](https://user-images.githubusercontent.com/48355355/164244427-ea376347-ab92-4d8d-b739-22dcf5f1372e.png)

**Comment by nathanlukeIDEA** [comment]
Hi Rayaan,

Please take a look here which describes how to add an 'added member' to OpenModel. 

[Connection Data - Added Member](https://github.com/idea-statica/ideastatica-public/wiki/IOM-Connection-Data)

Let me know if this does not help.

Regards
Nathan

**Comment by BouwenmetStaal** [comment]
Thanks Nathan,

Is ConnectionData not retrievable fom Openmodel?
This will save me defining an extra reference.

Example_AddAddedMember(OpenModel openModel, int openModelMemberId, ConnectionData connectionData)

**Comment by nathanlukeIDEA** [comment]
Hi Rayaan, 

[ConnectionData](https://github.com/idea-statica/ideastatica-public/blob/main/src/IdeaRS.OpenModel/Connection/ConnectionData.cs) is a class in OpenModel. Not to be confused with the namespace `IdeaRS.Connections.Data`.

The steel frame example explains adding beam data [here](https://github.com/idea-statica/ideastatica-public/tree/main/src/Examples/IOM#connected-members).

I have updated the example I shared on the wiki to make it a little clearer.

Cheers
Nathan

**Comment by BouwenmetStaal** [comment]
Thnx Nathan,

One new question, how can I modify the orientation (local, x,y,z) of the beam and its lengths (L1 and L2).
![image](https://user-images.githubusercontent.com/48355355/165086391-144983fb-9316-4666-8903-b52976ad00ad.png)

**Comment by nathanlukeIDEA** [comment]
Hi Rayaan,

You need to define the LineSegment3D (defines LCS) > Polyline3D > Element1D > Member1D (the same as you would any other member). Only when you add it to BeamData do you select the `IsAdded` = true` which basically tells IDEA to make it a stiffening operation, not a member. The LCS and length inputs in the UI are all automatically generated from the member definition - You don't need to define this separately anywhere in the code.

The best way to understand how to create what you are trying to achieve is to create it manually in IDEA Connection and then export it to IOM Model XML using developer Mode. Reading the XML It should make it pretty clear what you need to do in the OpenModel code.
